lms/djangoapps/courseware/views/index.py

```diff
@@ -446,6 +452,32 @@ def _create_courseware_context(self, request):
         )
         staff_access = self.is_staff
 
+        reset_deadlines_url = reverse(
+            'openedx.course_experience.reset_course_deadlines', kwargs={'course_id': six.text_type(self.course.id)}
+        )
+
+        allow_anonymous = allow_public_access(self.course, [COURSE_VISIBILITY_PUBLIC])
+
+        display_reset_dates_banner = False
+        if not allow_anonymous:  # pylint: disable=too-many-nested-blocks
+            course_overview = CourseOverview.objects.get(id=str(self.course_key))
+            end_date = getattr(course_overview, 'end_date')
+            if not end_date or timezone.now() < getattr(course_overview, 'end_date'):
+                course_enrollment = CourseEnrollment.objects.filter(course=course_overview, user=request.user).first()
+                if course_enrollment and course_enrollment.mode == CourseMode.VERIFIED:
```

Comment:
Would it be worth combining these two checks (for `end_date` already having passed, and having a verified enrollment) into a separate function somewhere, so that you can use it both here and in  `lms/djangoapps/courseware/views/views.py`?